### PR TITLE
Fix story card height for taller videos

### PR DESCRIPTION
**Pull request by gkop**
Also fix bug with audio and video component ids, and bump
react-virtualized

https://github.com/rubyforgood/terrastories/issues/17

### CODE REVIEW

File: rails/app/javascript/components/StoryMedia.jsx [hunk 1]
```diff
@@ -1,25 +1,35 @@
...
+    this.state = {
+      explicitVideoHeight: null
+    }
+  }
+
   renderAudio() {
     const { file } = this.props;
     return (
-      <audio id={`audio-player${file.id}`} controls>
+      <audio id={`audio-player${file.blob.id}`} controls>
```
Comment (gkop) [inline rails/app/javascript/components/StoryMedia.jsx]: fixing a bug I noticed incidentally.

File: rails/app/javascript/components/StoryMedia.jsx [hunk 2]
```diff
@@ -1,25 +1,35 @@
...
     );
   }
 
   renderVideo() {
     const { file } = this.props;
+    const { explicitVideoHeight } = this.state;
     return (
       <video
-        id={`video-player${file.id}`}
+        id={`video-player${file.blob.id}`}
```
Comment (gkop) [inline rails/app/javascript/components/StoryMedia.jsx]: fixing a bug I noticed incidentally.

File: rails/package.json [hunk 3]
```diff
@@ -8,8 +8,8 @@
     "react": "^16.4.2",
     "react-dom": "^16.4.2",
     "react-select": "^2.0.0",
+    "react-virtualized": "9.21.0",
```
Comment (gkop) [inline rails/package.json]: bumped this to see if it helped fix the issue. It did not. It's a small bump so maybe OK? happy to take out if it's too much in one patch.

File: rails/yarn.lock [hunk 4]
```diff
@@ -9,6 +9,13 @@
   dependencies:
     "@babel/types" "^7.0.0"
 
+"@babel/runtime@^7.1.2":
+  version "7.4.2"
+  resolved "https://registry.yarnpkg.com/@babel/runtime/-/runtime-7.4.2.tgz#f5ab6897320f16decd855eed70b705908a313fe8"
+  integrity sha512-7Bl2rALb7HpvXFL7TETNzKSAeBVCPHELzc0C//9FCxN8nsiueWSJBqaF+2oIJScyILStASR/Cx5WMkXGYTiJFA==
+  dependencies:
+    regenerator-runtime "^0.13.2"
```
Comment (gkop) [inline rails/yarn.lock]: I don't know why all this metadata got added :/

File: rails/app/javascript/components/StoryList.jsx [hunk 5]
```diff
@@ -49,6 +49,9 @@ class StoryList extends Component {
   renderStory = ({key, index, style, parent}) => {
     const story = this.props.stories[index];
     const storyClass = this.state.activeStoryIndex === index ? `story${index} isActive` : `story${index}`;
+    const bustCache = () => {
+      this.cache.clear(index, 0);
```
Comment (gkop) [inline rails/app/javascript/components/StoryList.jsx]: It is necessary I believe to bust this cache once we've loaded the first frame of the video and set the explicit height, in order to hint to the virtual list dynamic layout to do its thing.

**Comment by gkop** [comment]
I feel pretty good about this general approach. I'd love feedback on the React and ES6 implementation details, as this isn't a strong area for me.

Also sadly there's still a small glitch where the virtual list dynamic layout doesn't readjust until you scroll more than zero, I don't understand why. But the UX is much better with this patch IMO.

File: rails/package.json (hunk 3 above)
Comment (baweaver) [reply rails/package.json]: Fine by me

File: rails/yarn.lock (hunk 4 above)
Comment (baweaver) [reply rails/yarn.lock]: Might just be the version cascade off of that package

**Pull request merged**